### DISCUSSION TITLE
Bug: Can't Route by page custom templates. EX: About, Contact, etc..

### DISCUSSION THREAD

**Discussion by JoeStantonCode** [wpengine/faustjs#1314] [Q&A]
Does this codebase support routing by custom template names? Right now it appears all the routes are based upon ContentType->Node->Name. 

Is there a way to also handle routes by ContentType->Template->TemplateName? Right now it appears all the templates will just use the page node name.

WordPress Template:

about.php
/* Template Name: About */

contact.php
/* Template Name: Contact */ 

and then within the wp-templates->index.js we would have something along these lines? 
About: about,
Contact: contact

Let me know if this is currently possible or if this would be a future feature? It'd be really nice to be able to route by custom template names so we can serve up different layouts and queries for about, contact, team, etc. Thanks! Great code base here.

**Comment by theodesp** [comment]
Hey @JoeStantonCode. You can use the [possibleTemplatesList](https://faustjs.org/docs/plugin-system/filters#possibletemplateslist) hook to add you own template resolution. You can consume the `seedNode` second parameter passed and add a different template to resolve:

```js
export class TemplateNameResolverPlugin {
  constructor() {}

  apply(hooks) {
    hooks.addFilter('possibleTemplatesList', 'faust', (templates, data) => {
      const templateName = data?.seedNode?.template?.templateName;
      if (templateName && templateName !== 'Default') {
        return Array.from(new Set([templateName, ...templates]));
      }
      return templates;
    });
  }
}
```
